modin/core/execution/ray/implementations/pandas_on_ray/partitioning/virtual_partition.py

```diff
@@ -493,17 +515,24 @@ def deploy_ray_func(func, *args):  # pragma: no cover
         Function to be executed on an axis partition.
     *args : iterable
         Additional arguments that need to passed in ``func``.
+    **kwargs : dict
+        Additional keyword arguments to be passed in `func`.
 
     Returns
     -------
-    list
+    list : Union[tuple, list]
         The result of the function ``func`` and metadata for it.
 
     Notes
     -----
     Ray functions are not detected by codecov (thus pragma: no cover).
     """
-    result = func(*args)
+    func = deserialize(func)
+    args = deserialize(args)
+    kwargs = deserialize(kwargs)
+    if "args" in kwargs:
+        kwargs["args"] = deserialize(kwargs["args"])
```

Comment:
Since this is a quick fix, I think it's best to leave this for now than rewrite `deserialize` to do something it doesn't need to do. The longer term fix https://github.com/modin-project/modin/pull/4473 will make this if statement unnecessary.